srcs/app/http/refactor/HttResponseFactory.cpp

```diff
@@ -0,0 +1,97 @@
+#include "HttpResponseFactory.hpp"
+
+HttpResponseFactory::HttpResponseFactory(
+		const std::string raw_request,
+		const ServerConfig &server_config
+	): server_config_(server_config) {
+	request_ = new HttpRequestDTO(raw_request);
```

Comment:
This part uses the old definition of the `HttpRequest` class, which constructs a request from a string (and expects it to be complete). The version we have in the main branch is a bit different.

If we want to update it to be the same as the main, maybe we don't need this variable, and we can add the request pointer as an argument of the `createHttp*Response_()`.

Reply:
Totally right about this too! thanks for pointing out :)  As said in previous comment, this is just a scaffolding, just to see how classes could relate to each other. This `HttpRequest` has been updated to new type in next PRs :)